docs/src/image_loading.md

```diff
@@ -0,0 +1,203 @@
+
+The Image Loading module provides the interface to retrieve and parse the manifest from the flash storage, and transfer SOC images from the storage to the target destination.
+
+The MCU SDK exposes APIs to authorize the manifest and verify images through the "Attestation", "Anti-Rollback Protection" and "Crypto" modules.
+
+### Application Interfaces
+
+The APIs are presented as methods of the ImageLoader struct.
+
+```
+/// Image loader module
+pub struct ImageLoader;
+
+impl ImageLoader {
+    /// APIs
+}
+```
+
+#### Get SOC Manifest
+
+```rust
+/// Retrieves the SOC manifest from flash memory.
+///
+/// # Returns
+/// The SOC manifest if it is successfully read from flash.
+pub fn get_soc_manifest(&self) -> Result<SocManifest, DynError>;
+```
+
+#### Get Image Metadata Entry
+
+```rust
+/// Retrieves an Image Metadata Entry by `firmware_id`.
+///
+/// # Arguments
+/// - `firmware_id`: The unique identifier for the image.
+///
+/// # Returns
+/// The Image Metadata Entry if it exists, or an error if it is not found.
+pub fn get_image_metadata(firmware_id: u32) -> Result<ImageMetadataEntry,DynError>;
+```
+
+#### Get Image Digest
+
+```rust
+    /// Retrieves the digest of an image by `firmware_id`.
+    ///
+    /// # Arguments
+    /// - `firmware_id`: The unique identifier for the image.
+    ///
+    /// # Returns
+    /// The digest (`[u8; 48]`) of the image if it exists, or an error if it is not found.
+    pub fn get_image_digest(image_identifier: u16) -> Result<[u8; 48], DynError>;
+```
+
+#### Load Image with DMA
+
+```rust
+    /// Loads an image from a source address to a destination address using DMA
+    /// This is a non-blocking operation, and the completion of the transfer is notified via a callback function.
+    ///
+    /// # Arguments
+    /// - `source_address`: The starting address of the image in flash memory.
+    /// - `destination_address`: The memory address where the image will be loaded.
+    /// - `size`: The size of the image to transfer in bytes.
+    /// - `callback`: A function to handle the status of the transfer.
+    ///
+    /// # Callback
+    /// The callback is a function pointer or closure of type:
+    /// ```rust
+    /// fn(status: DmaStatus);
+    /// ```
+    /// Where `DmaStatus` indicates the completion or failure status of the DMA transfer.
+    pub fn load_image_with_dma(
+        &self,
+        source_address: usize,
+        destination_address: usize,
+        size: usize,
+        callback: fn(status: DmaStatus),
+    );
+```
+
+## Using Image Loading in the Application
+
+This section describes how to use the image_loader module to load an image from flash into its designated destination memory. The application interacts with the image_loader and the security modules to verify and authorize the image before loading it.
```

Comment:
Try to be consistent with how we refer to the module. Maybe "Image Loading module" here.

If we are referring to a Rust package, we can say `image_loader` (in monospaced text), or a `trait` can also be monospaced, like `ImageLoader`.